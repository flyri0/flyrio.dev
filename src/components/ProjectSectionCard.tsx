import TagList from '@components/TagList';

interface Props {
  tags: String[];
  image: {
    src: string;
    alt: string;
  };
  title: string;
  description: string;
  path: string;
}

export default function ProjectCard({
  tags,
  image,
  title,
  description,
  path,
}: Props): JSX.Element {
  return (
    <a
      href={path}
      className="group flex flex-col gap-y-3 rounded border border-accent-2 p-5 font-spaceGrotesk transition hover:bg-accent-1">
      <h1 className="text-3xl font-bold group-hover:text-highlight group-hover:underline">
        {title}
      </h1>
      <p className="font-bold text-shadow">{description}</p>
      <TagList tagsArray={tags} />
      <img className="rounded" src={`/projects/${image.src}`} alt={image.alt} />
    </a>
  );
}
