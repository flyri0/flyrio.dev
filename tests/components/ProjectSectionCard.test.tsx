import { render, screen } from '@testing-library/react';

import ProjectSectionCard from '@components/ProjectSectionCard';

describe('ProjectSectionCard', () => {
  const TAGS = ['Python', 'AWS', 'TypeScript'];
  const IMAGE = { src: 'example.png', alt: 'example image' };
  const TITLE = 'Awesome Title';
  const DESCRIPTION = 'The Best Description Ever';
  const PATH = '/best-post';

  test('renders correctly', () => {
    const element = render(
      <ProjectSectionCard
        tags={TAGS}
        image={IMAGE}
        title={TITLE}
        description={DESCRIPTION}
        path={PATH}
      />
    );

    expect(element).toMatchSnapshot();
  });

  test('should navigate to /best-post when link is clicked', () => {
    render(
      <ProjectSectionCard
        tags={TAGS}
        image={IMAGE}
        title={TITLE}
        description={DESCRIPTION}
        path={PATH}
      />
    );

    expect(screen.getByRole('link')).toHaveAttribute('href', '/best-post');
  });
});
