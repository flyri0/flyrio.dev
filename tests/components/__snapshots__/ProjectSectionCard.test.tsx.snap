// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProjectSectionCard > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="group flex flex-col gap-y-3 rounded border border-accent-2 p-5 font-spaceGrotesk transition hover:bg-accent-1"
        href="/best-post"
      >
        <h1
          class="text-3xl font-bold group-hover:text-highlight group-hover:underline"
        >
          Awesome Title
        </h1>
        <p
          class="font-bold text-shadow"
        >
          The Best Description Ever
        </p>
        <div
          class="flex items-start gap-x-1 text-xs font-bold"
        >
          <span
            class="rounded border border-shadow p-1 text-shadow"
          >
            Python
          </span>
          <span
            class="rounded border border-shadow p-1 text-shadow"
          >
            AWS
          </span>
          <span
            class="rounded border border-shadow p-1 text-shadow"
          >
            TypeScript
          </span>
        </div>
        <img
          alt="example image"
          class="rounded"
          src="/projects/example.png"
        />
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="group flex flex-col gap-y-3 rounded border border-accent-2 p-5 font-spaceGrotesk transition hover:bg-accent-1"
      href="/best-post"
    >
      <h1
        class="text-3xl font-bold group-hover:text-highlight group-hover:underline"
      >
        Awesome Title
      </h1>
      <p
        class="font-bold text-shadow"
      >
        The Best Description Ever
      </p>
      <div
        class="flex items-start gap-x-1 text-xs font-bold"
      >
        <span
          class="rounded border border-shadow p-1 text-shadow"
        >
          Python
        </span>
        <span
          class="rounded border border-shadow p-1 text-shadow"
        >
          AWS
        </span>
        <span
          class="rounded border border-shadow p-1 text-shadow"
        >
          TypeScript
        </span>
      </div>
      <img
        alt="example image"
        class="rounded"
        src="/projects/example.png"
      />
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
